packages/types/src/ts-estree.ts

```diff
@@ -1176,6 +1178,11 @@ export interface ObjectPattern extends BaseNode {
   decorators?: Decorator[];
 }
 
+export interface ParenthesizedExpression extends BaseNode {
+  type: AST_NODE_TYPES.ParenthesizedExpression;
```

Comment:
Because this case is *specifically* about JSDoc type assertions, and we have no immediate or future plans to support more parenthesized expressions - I think we should call this `JSDocTypeAssertion`.

Reply:
`ParenthesizedExpression` is an established thing in the ecosystem. Could we keep this `type` and add a boolean property `jsDocTypeAnnotation: true` instead? (The `typeAnnotation` property will do too, but it's out of scope for now I guess.)

Reply:
Within ESTree its not an established practice. Most JS parsers don't emit it either. Babel does emit it all-or-nothing via a flag.

Adding a new node type is a much better API than using something generic because it's a special type of expression which has special meaning AND syntax to typescript and we should treat it akin to an `as` or angle bracket assertion. This isn't just a regular parenthesised expression where the parentheses control order of operations.

Reply:
see https://github.com/babel/babel/pull/12933#pullrequestreview-617676009

Reply:
> Babel does emit it all-or-nothing via a flag.

I'd say that's what most JS parsers do. I was a bit worried about compatibility issues for various tools like recast. There is no evidence people use it with this parser though. If you think it's not a big deal, so be it.

Reply:
@armano2 Note that that PR doesn't change whether or not we create `ParentesizedExpression` nodes, it only ensures that we don't throw when we do emit them.